service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseInitiationService.java

```diff
@@ -187,10 +183,34 @@ public CaseData updateOrganisationsDetails(CaseData caseData) {
             return addCourtDetails(updatedCaseData);
         }
 
-        throw new IllegalStateException("Cannot determine local authority for a case");
+        if (nonNull(caseData.getRepresentativeType())) {
+            boolean isRespondentSolicitor = Objects.equals(caseData.getRepresentativeType().toString(),
+                "RESPONDENT_SOLICITOR");
+            boolean isChildSolicitor = Objects.equals(caseData.getRepresentativeType().toString(),
+                "CHILD_SOLICITOR");
+
+            if (isRespondentSolicitor || isChildSolicitor) {
+                CaseData updatedCaseData = caseData.toBuilder()
+                    .outsourcingPolicy(organisationPolicy(
+                        currentUserOrganisationId,
+                        currentUserOrganisationName,
+                        isRespondentSolicitor ? SOLICITORA : CHILDSOLICITORA))
+                    .build();
+
+                return addCourtDetails(updatedCaseData);
+            }
+        }
+
+        return caseData;
     }
 
     private CaseData addCourtDetails(CaseData caseData) {
+        if (Objects.equals(caseData.getCaseLocalAuthorityName(), null)) {
```

Comment:
`isNotEmpty`

Reply:
also does this block mean that if a respondent/child solicitor is creating a case there is no designated court? (or is that still to be chosen later in Orders + Directions sought?)

Reply:
So for LA's the court is set here from the mappings, but for sols no such mapping exists so its set to null for now and yes you're correct in that its set in the Orders + Directions sought stage